src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.Analyzers.Common/SyntaxNodeExtensions.cs

```diff
@@ -0,0 +1,212 @@
+        /// Applies whitespace trivia and new line trivia from another syntax node to this one.
+        /// </summary>
+        /// <typeparam name="T">The type of syntax node to be updated.</typeparam>
+        /// <param name="statement">The syntax node to update.</param>
+        /// <param name="otherStatement">The syntax node to copy trivia from.</param>
+        /// <returns>The original syntax node updated with the other syntax's whitespace and new line trivia.</returns>
+        public static T WithWhitespaceTriviaFrom<T>(this T statement, SyntaxNode otherStatement)
+            where T : SyntaxNode
+        {
+            return statement
+                .WithLeadingTrivia(otherStatement?.GetLeadingTrivia().Where(IsWhitespaceTrivia) ?? SyntaxTriviaList.Empty)
+                .WithTrailingTrivia(otherStatement?.GetTrailingTrivia().Where(IsWhitespaceTrivia) ?? SyntaxTriviaList.Empty);
+
+            static bool IsWhitespaceTrivia(SyntaxTrivia trivia) =>
+                CSharpExtensions.IsKind(trivia, CS.SyntaxKind.EndOfLineTrivia)
+                || CSharpExtensions.IsKind(trivia, CS.SyntaxKind.WhitespaceTrivia)
+                || VisualBasicExtensions.IsKind(trivia, VB.SyntaxKind.EndOfLineTrivia)
+                || VisualBasicExtensions.IsKind(trivia, VB.SyntaxKind.WhitespaceTrivia);
+        }
+
+        /// <summary>
+        /// Determines if a syntax node includes a using or import statement for a given namespace.
+        /// </summary>
+        /// <param name="node">The node to analyze.</param>
+        /// <param name="namespaceName">The namespace name to check for.</param>
+        /// <returns>True if the node or any of its descendents have a an import or using statement for the given namespace. False otherwise.</returns>
+        public static bool HasUsingStatement(this SyntaxNode node, string namespaceName)
+        {
+            if (node is null)
+            {
+                throw new ArgumentNullException(nameof(node));
+            }
+
+            var nodes = node.DescendantNodesAndSelf(_ => true, false);
+
+            return nodes.OfType<CSSyntax.UsingDirectiveSyntax>().Any(u => u.Name.ToString().Equals(namespaceName, StringComparison.Ordinal))
+                || nodes.OfType<VBSyntax.SimpleImportsClauseSyntax>().Any(i => i.Name.ToString().Equals(namespaceName, StringComparison.Ordinal));
+        }
+
+        /// <summary>
+        /// Adds a using directive for a given namespace to the document root only if the directive is not already present.
+        /// </summary>
+        /// <param name="documentRoot">The document to add the directive to.</param>
+        /// <param name="namespaceName">The namespace to reference with the using directive.</param>
+        /// <returns>An updated document root with the specific using directive.</returns>
+        public static CSSyntax.CompilationUnitSyntax AddUsingIfMissing(this CSSyntax.CompilationUnitSyntax documentRoot, string namespaceName)
```

Comment:
does this have to be CSharp aware?

Reply:
Ideally, this will go away and be replaced by `ImportAdder`. It's only here temporarily because it was previously used by some code fix providers that haven't been updated to be cross-language yet (HttpContext.Current and HttpContext.IsDebuggingEnabled). I'll put a comment noting that it should be removed and address it in a separate PR.